libcaf_core/caf/string_algorithms.test.cpp

```diff
@@ -0,0 +1,83 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/string_algorithms.hpp"
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/test.hpp"
+
+#include <string>
+#include <vector>
+
+using namespace caf;
+
+namespace {
+
+using str_list = std::vector<std::string>;
+
+str_list split(std::string_view str) {
+  str_list result;
+  caf::split(result, str, ",");
+  return result;
+}
+
+str_list compressed_split(std::string_view str) {
+  str_list result;
+  caf::split(result, str, ",", token_compress_on);
+  return result;
+}
+
+std::string join(str_list vec) {
+  return caf::join(vec, ",");
+}
+
+TEST("splitting") {
+  check_eq(split(""), str_list({""}));
```

Comment:
```
  check_eq(split(""), str_list{""});
```

We've used the extra parens to protect against the CHECK_EQ macro before, so they're no longer needed. Same in the test below